libsolidity/codegen/ir/IRGeneratorForStatements.h

```diff
@@ -93,7 +106,12 @@ class IRGeneratorForStatements: public ASTConstVisitor
 		std::string const& _right
 	);
 
-	void setLValue(Expression const& _expression, std::unique_ptr<IRLValue> _lvalue);
+	/// Assigns the value of @a _value to the current lvalue. Resets the current lvalue.
+	void assignCurrentLValue(IRVariable const& _value);
```

Comment:
I'm not too happy about the "current" in both function names. `fetchCurrentLValue` is basically "use lvalue in rvalue context", or "dereference". Also from the name "assignCurrentLValue" and "setLValue" sound really similar...

Reply:
I'm not too happy with the lvalue stuff in general... but getting rid of it entirely would have meant even more changes and I thought this is huge enough already... but I can try to come up with better names in any case - maybe even just ``void writeToLValue(IRLValue const& _lvalue, IRVariable _value)`` and ``IRVariable readFromLValue(IRLValue const& _lvalue)`` having the caller pass in ``m_currentLValue``...